### PR TITLE
Properly close multiple idle connections

### PR DESCRIPTION
**Pull request by masaruh**
Fixes an issue where it doesn't close connections even after client_inactivity_timeout
has passed when there are multiple connections.

### CODE REVIEW

**Comment by ph** [comment]
Fix the test run I've applied this patch and this should fix any travis errors, we did some changes upstream that make the load order more important.

```diff
diff --git a/spec/inputs/beats/codec_callback_listener_spec.rb b/spec/inputs/beats/codec_callback_listener_spec.rb
index 5c08b57..9329f04 100644
--- a/spec/inputs/beats/codec_callback_listener_spec.rb
+++ b/spec/inputs/beats/codec_callback_listener_spec.rb
@@ -1,7 +1,7 @@
 # encoding: utf-8
+require_relative "../../spec_helper"
 require "logstash/inputs/beats"
 require "logstash/event"
-require "spec_helper"
 require "thread"
 
 describe LogStash::Inputs::Beats::CodecCallbackListener do
diff --git a/spec/spec_helper.rb b/spec/spec_helper.rb
index 67e3a42..7ec54aa 100644
--- a/spec/spec_helper.rb
+++ b/spec/spec_helper.rb
@@ -1,8 +1,8 @@
 # encoding: utf-8
+require "logstash/devutils/rspec/spec_helper"
 require "rspec"
 require "rspec/mocks"
 require "rspec/wait"
-require "logstash/devutils/rspec/spec_helper"
 require "logstash/codecs/plain"
 require_relative "support/logstash_test"
 require_relative "support/helpers"

```

File: src/test/java/org/logstash/beats/ServerTest.java [hunk 1]
```diff
@@ -52,18 +94,20 @@ public void run() {
...
-
         try {
             Long started = System.currentTimeMillis() / 1000L;
 
-            connectClient().sync().channel().closeFuture().sync();
+
+            for (int i = 0; i < concurrentConnections; i++) {
+                connectClient();
+            }
+            latch.await(10, TimeUnit.SECONDS);
```
Comment (ph) [inline src/test/java/org/logstash/beats/ServerTest.java]: TIL

**Comment by ph** [comment]
@masaruh I've tested this PR and I still see connection in the netstat, long after the timeout period.
I wonder if something else prevent the timeout event to be executed.

**Comment by masaruh** [comment]
Hmm, it works for me... Making it not sharable should work too.

Since #191 supersedes this issue, I'm going to close this issue. But pushing new change which makes it not have state, just in case...

**Comment by ph** [comment]
@masaruh I agree I would prefer not having to keep a state, I have no other indication that the queue is blocked at the moment and this plugin need to work in both 2.x 5.x :(

**Pull request closed without merging**